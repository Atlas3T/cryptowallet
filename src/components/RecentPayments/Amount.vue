<!--

it renders a div with amount-ox class
it variant grey is passed, it renders grey-text class
it rendersp properly formatted amount

 -->


<template>
  <div
    :class="{ 'grey-text': variant.indexOf('grey') != -1 }"
    class="amount-box"
  >
    <div>{{ formattedAmount }}</div>
  </div>
</template>

<script>
import numeral from 'numeral';

export default {
  name: 'Amount',
  props: {
    amount: {
      type: Number,
      required: true,
    },
    variant: {
      type: String,
      default() {
        return '';
      },
    },
  },
  computed: {
    formattedAmount() {
      const formatted = `${numeral(this.amount).format('0.00')} Â£`;
      return (this.amount > 0 ? `+${formatted}` : formatted);
    },
  },
};
</script>

<style scoped>
  .amount-box {
    display: flex;
    justify-content: flex-end;
    text-align: right;
    white-space: nowrap;
  }
</style>
