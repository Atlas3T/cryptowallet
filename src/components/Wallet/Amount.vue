<template>
  <div
    :class="{ 'grey-text': variant.indexOf('grey') >= 0 }"
    class="amount-box"
  >
    <div>{{ formattedAmount }}</div>
  </div>
</template>

<script>
import numeral from 'numeral';

export default {
  name: 'Amount',
  props: {
    amount: {
      type: Number,
      required: true,
    },
    variant: {
      type: String,
      default() {
        return '';
      },
    },
  },
  computed: {
    /*
     * converts raw amount {Number} into a {String} that begins with
     * either "+" or "-" and ends with " £"" also, it formats the number
     * into currency friendly format: "0.00"
     * @return {String} eg. "+12.40 £" or "-1.00 £"
     */
    formattedAmount() {
      const formatted = `${numeral(this.amount).format('0.00')} £`;
      return (this.amount > 0 ? `+${formatted}` : formatted);
    },
  },
};
</script>

<style scoped>
  .amount-box {
    display: flex;
    justify-content: flex-end;
    text-align: right;
    white-space: nowrap;
  }
</style>
