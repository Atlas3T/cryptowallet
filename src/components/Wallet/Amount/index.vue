<template>
  <div class="amount-box">
    {{ formattedAmount }}
  </div>
</template>
<script>
import AmountFormatter from '@/helpers/AmountFormatter';

export default {
  name: 'Amount',
  props: {
    amount: {
      type: Number,
      required: true,
    },
    rate: {
      type: String,
      required: false,
    },
    prependPlusOrMinus: {
      type: Boolean,
      required: false,
    },
    currency: {
      type: Object,
      required: false,
    },
    toCurrency: {
      type: Boolean,
      required: false,
    },
    toCoin: {
      type: Boolean,
      required: false,
    },
    coin: {
      type: String,
      required: false,
    },
    format: {
      type: String,
      required: false,
    },
  },
  computed: {
    /**
     * Converts amount value (Number) into a String
     * @return {String}
     */
    formattedAmount() {
      const {
        amount,
        rate,
        format,
        coin,
        prependPlusOrMinus,
        currency,
        toCurrency,
        toCoin,
      } = this;

      const formattedAmount = new AmountFormatter({
        amount,
        rate,
        format,
        coin,
        prependPlusOrMinus,
        currency,
        toCurrency,
        toCoin,
        withCurrencySymbol: true,
      });

      return formattedAmount.getFormatted();
    },
  },
};
</script>

<style scoped>
  .amount-box {
    display: flex;
    white-space: nowrap;
  }
</style>
