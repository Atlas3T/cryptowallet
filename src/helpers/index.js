import AmountFormatter from './AmountFormatter';
import dateTranslater from './DateTranslater';
import Contacts from './Contacts';

export {
  AmountFormatter,
  dateTranslater,
  Contacts,
};
