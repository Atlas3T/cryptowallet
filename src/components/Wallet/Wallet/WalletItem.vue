<template>
  <div
    class="coin"
    @click="selectCoin(data.uid)"
  >
    <div class="details">
      <div class="icon"/>
      <div class="name">{{ data.name }}</div>
    </div>
    <div class="balance">
      <div>{{ data.balance }}</div>
      <div>
        <Amount
          :amount="data.balance"
          :prepend-plus-or-minus="false"
          :currency="currency"
          :coin="data.key"
        />
      </div>
    </div>
    <div>
      <q-icon name="chevron_right"/>
    </div>
  </div>
</template>

<script>
import Amount from '@/components/Wallet/SharedComponents/Amount.vue';

export default {
  name: 'WalletItem',
  components: {
    Amount,
  },
  props: {
    data: {
      type: Object,
      required: true,
    },
    currency: {
      type: String,
      required: true,
    },
  },
  methods: {
    selectCoin(uid) {
      this.$router.push({ path: `/wallet/balance/${uid}` });
    },
  },
};
</script>

<style scoped>
.coin {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
  border-bottom: 1px solid #13273a;
}

.coin .details {
  display: flex;
  align-items: center;
}

.icon {
  border-radius:100%;
  background: #41678a;
  width: 2.5rem;
  height: 2.5rem;
  margin-right: 1rem;
}
</style>
