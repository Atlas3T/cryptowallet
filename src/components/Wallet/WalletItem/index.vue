<template>
  <div
    class="wallet selectWallet"
    @click="selectWallet(wallet.uid)"
  >
    <div class="details">
      <div class="icon"/>
      <div class="name">{{ wallet.name }}</div>
    </div>
    <div class="balance">
      <div>{{ wallet.balance }}</div>
      <Amount
        :amount="wallet.balance"
        :prepend-plus-or-minus="false"
        :currency="currency"
        :coin="wallet.key"
      />
    </div>
    <div>
      <q-icon name="chevron_right"/>
    </div>
  </div>
</template>

<script>
import Amount from '@/components/Wallet/Amount';

export default {
  name: 'WalletItem',

  components: {
    Amount,
  },

  props: {
    wallet: {
      type: Object,
      required: true,
    },
    currency: {
      type: Object,
      required: true,
    },
  },

  methods: {
    selectWallet(uid) {
      this.$router.push({ path: `/wallet/balance/${uid}` });
    },
  },
};
</script>

<style scoped>
.coin {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
  border-bottom: 1px solid #13273a;
}

.coin .details {
  display: flex;
  align-items: center;
}

.icon {
  border-radius:100%;
  background: #41678a;
  width: 2.5rem;
  height: 2.5rem;
  margin-right: 1rem;
}
</style>
