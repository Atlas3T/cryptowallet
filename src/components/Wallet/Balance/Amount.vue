<template>
  <div
    :class="{ 'grey-text': variant.indexOf('grey') >= 0 }"
    class="amount-box"
  >
    <div>{{ formattedAmount }}</div>
  </div>
</template>

<script>
import { amount } from '@/helpers';

export default {
  name: 'Amount',
  props: {
    amount: {
      type: Number,
      required: true,
    },
    variant: {
      type: String,
      default() {
        return '';
      },
    },
    currency: {
      type: String,
      default() {
        return '';
      },
    },
  },
  computed: {
    /**
     * Converts amount value (Number) into a String with optional currency symbol
     * @return {String}
     */
    formattedAmount() {
      return amount(this.amount, this.currency);
    },
  },
};
</script>

<style scoped>
  .amount-box {
    display: flex;
    justify-content: flex-end;
    text-align: right;
    white-space: nowrap;
  }
</style>
