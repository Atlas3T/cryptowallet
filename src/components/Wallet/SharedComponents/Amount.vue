<template>
  <div
    class="amount-box"
  >
    <div v-html="formattedAmount"/>
  </div>
</template>

<script>
import { Amount } from '@/helpers';

export default {
  name: 'Amount',
  props: {
    amount: {
      type: Number,
      required: true,
    },
    prependPlusOrMinus: {
      type: Boolean,
      default() {
        return true;
      },
    },
    currency: {
      type: String,
      default() {
        return null;
      },
    },
    coin: {
      type: String,
      default() {
        return null;
      },
    },
    format: {
      type: String,
      default() {
        return null;
      },
    },
  },
  computed: {
    /**
     * Converts amount value (Number) into a String
     * @return {String}
     */
    formattedAmount() {
      const {
        amount,
        format,
        coin,
        prependPlusOrMinus,
        currency,
      } = this;

      const formattedAmount = new Amount({
        amount,
        format,
        coin,
        prependPlusOrMinus,
        currency,
      });

      return formattedAmount.formatted;
    },
  },
};
</script>

<style scoped>
  .amount-box {
    display: flex;
    white-space: nowrap;
  }
</style>
