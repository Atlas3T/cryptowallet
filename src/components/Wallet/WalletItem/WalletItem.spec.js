import { shallowMount } from '@vue/test-utils';
import WalletItem from '@/components/Wallet/WalletItem';
import Amount from '@/components/Wallet/Amount';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';

describe('WalletItem.vue', () => {
  let storeMocks;
  let wrapper;
  let store;
  let router;

  const propsData = {
    data: {
      key: 'BTC',
      name: 'Bitcoin',
      balance: 25.532344,
      uid: 1,
    },
    currency: {},
    wallet: {
      balance: 10,
    },
  };

  const walletObjProp = {
    wallet: {
      uid: 1
    }
  };

  function wrapperInit (options) {
    return shallowMount(WalletItem, options);
  }

  function storeInit (custom) {
    storeMocks = createStoreMocks(custom);
    router = createRouter(storeMocks.store);
    router.push({ path: `/` });
    wrapper = wrapperInit({ i18n, router, localVue, store: storeMocks.store, propsData });
    store = wrapper.vm.$store;
  }

  beforeEach(() => storeInit());

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('calls selectWalletMock() method on click correctly', () => {
    const selectWalletMock = jest.fn();
    wrapper.setMethods({selectWallet: selectWalletMock});
    wrapper.setProps(walletObjProp);
    wrapper.find('.selectWallet').trigger('click');
    expect(selectWalletMock).toHaveBeenCalledWith(1);
  });

  it('renders the coin name', () => {
    expect(wrapper.html().includes(propsData.data.name));
  });

  it('renders the balance', () => {
    expect(wrapper.html().includes(propsData.data.balance));
  });

  it('renders the Amount component', () => {
    expect(wrapper.contains(Amount)).toBe(true);
  });

  it('renders the icon components correctly', () => {
    expect(wrapper.contains('qicon-stub[name="chevron_right"]')).toBe(true);
  });

  it('navigates to correct route when clicked on', () => {
    expect(store.state.route.path).toBe('/');
    wrapper.setProps(walletObjProp);
    wrapper.find('.selectWallet').trigger('click');
    expect(store.state.route.path).toBe(`/wallet/balance/${propsData.data.uid}`);
  });
});
