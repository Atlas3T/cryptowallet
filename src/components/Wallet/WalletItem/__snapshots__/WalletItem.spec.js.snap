// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletItem.vue renders and matches snapshot 1`] = `
<div
  class="wallet selectWallet"
  data="[object Object]"
>
  <div
    class="details"
  >
    <div
      class="icon"
    />
     
    <div
      class="name"
    >
      
    </div>
  </div>
   
  <div
    class="balance"
  >
    <div>
      10
    </div>
     
    <amount-stub
      amount="10"
      currency="[object Object]"
    />
  </div>
   
  <div>
    <qicon-stub
      name="chevron_right"
    />
  </div>
</div>
`;
